File: APTExercise/app/src/main/java/me/tommy/apt_exercise/MainActivity.java
package me.tommy.apt_exercise;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import me.tommy.autoviewbind.AutoBind;
import me.tommy.autoviewbind.launcher.AutoBindView;

public class MainActivity extends AppCompatActivity {

    @AutoBind(R.id.tv_apt_test)
    public TextView tvAptTest;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        AutoBindView.getInstance().inject(this);

        tvAptTest.setText("APT 测试");
    }
}

File: JNI Sample/app/src/main/java/me/tommy/jnisample/MainActivity.kt
package me.tommy.jnisample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import me.tommy.jnisample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.sampleText.text = stringFromJNI()
        binding.sampleText2.text = stringFromJNI2()
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    external fun stringFromJNI2():String

    companion object {
        // Used to load the 'native-lib' library on application startup.
        init {
            System.loadLibrary("native-lib")
        }
    }
}
